internal/mode/static/config_updater.go

```diff
@@ -17,24 +19,43 @@ import (
 	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
 )
 
-// ZapLogLevelSetter defines an interface for setting the logging level of a zap logger.
-type ZapLogLevelSetter interface {
+// logLevelSetter defines an interface for setting the logging level of a logger.
+type logLevelSetter interface {
 	SetLevel(string) error
-	Enabled(zapcore.Level) bool
 }
 
-type zapSetterImpl struct {
+// multiLogLevelSetter sets the log level for multiple logLevelSetters.
+type multiLogLevelSetter struct {
+	setters []logLevelSetter
+}
+
+func newMultiLogLevelSetter(setters ...logLevelSetter) multiLogLevelSetter {
+	return multiLogLevelSetter{setters: setters}
+}
+
+func (m multiLogLevelSetter) SetLevel(level string) error {
+	for _, s := range m.setters {
+		if err := s.SetLevel(level); err != nil {
```

Comment:
would it make sense to set all setters and join any errors https://pkg.go.dev/errors#Join ?